packages/ui/src/badge/badge.tsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * WordPress dependencies
+ */
+import { forwardRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { Box } from '../box';
+import { type BoxProps } from '../box/types';
+import { type BadgeProps } from './types';
+
+/**
+ * Default render function that renders a span element with the given props.
+ *
+ * @param props The props to apply to the HTML element.
+ */
+const DEFAULT_RENDER = ( props: React.ComponentPropsWithoutRef< 'span' > ) => (
+	<span { ...props } />
+);
+
+/**
+ * Maps intent values to Box backgroundColor and color props.
+ * Uses strong emphasis styles (as emphasis prop has been removed).
+ * @param intent
+ */
+const getIntentStyles = (
+	intent: BadgeProps[ 'intent' ]
+): Pick<
+	BoxProps,
+	'backgroundColor' | 'color' | 'borderColor' | 'borderWidth'
+> => {
+	switch ( intent ) {
+		case 'high':
+			return {
+				backgroundColor: 'error' as const,
+				color: 'error' as const,
+			};
+		case 'medium':
+			return {
+				backgroundColor: 'warning' as const,
+				color: 'warning' as const,
+			};
+		case 'low':
+			return {
+				backgroundColor: 'caution' as const,
+				color: 'caution' as const,
+			};
+		case 'stable':
+			return {
+				backgroundColor: 'success' as const,
+				color: 'success' as const,
+			};
+		case 'informational':
+			return {
+				backgroundColor: 'info' as const,
+				color: 'info' as const,
+			};
+		case 'draft':
+			return {
+				backgroundColor: 'neutral-weak' as const,
+				color: 'neutral' as const,
+			};
+		case 'none':
+		default:
+			return {
+				backgroundColor: 'neutral-strong' as const,
+				color: 'neutral' as const,
+				borderColor: 'neutral' as const,
+				borderWidth: 'xs' as const,
```

Comment:
I don't think we need these `as const`. The `Pick` in the function return value should already be doing the work of communicating that these are valid values for the Box props, and not just generic `string` type.

```suggestion
				backgroundColor: 'error',
				color: 'error',
			};
		case 'medium':
			return {
				backgroundColor: 'warning',
				color: 'warning',
			};
		case 'low':
			return {
				backgroundColor: 'caution',
				color: 'caution',
			};
		case 'stable':
			return {
				backgroundColor: 'success',
				color: 'success',
			};
		case 'informational':
			return {
				backgroundColor: 'info',
				color: 'info',
			};
		case 'draft':
			return {
				backgroundColor: 'neutral-weak',
				color: 'neutral',
			};
		case 'none':
		default:
			return {
				backgroundColor: 'neutral-strong',
				color: 'neutral',
				borderColor: 'neutral',
				borderWidth: 'xs',
```